tests/eas/preliminary: WorkThroughput improvement

valschneider opened the pull request:
_check_work_throughput() had an unused 'margin' variable. It's now used
to check that each CPU does at least 10% more work at its highest OPP
than at its lowest. This also 'fixes' false positives induced by the
fact that every sysbench run is not quite identical, so we could be
comparing 99 events vs 100 events, which should be a failure.

Also added a list of (freq -> work) in the test output should it fail,
which can help figuring out what is causing the failure

valschneider commented:
Arguably the (freq -> work) list isn't **that** useful, and can be quite big if there are a lot of OPPs on the target. We could just list the lowest & highest OPPs, but since the test relies on them it would simply be easier to print "work at highest OPP wasn't 10% bigger than work at lowest OPP".

bjackman commented on tests/eas/preliminary.py:
```diff
@@ -133,37 +133,55 @@ def _run_sysbench_work(self, cpu, duration):
+        for cpu in failed_results.iterkeys():
+            msg += '\nCPU {}:'.format(cpu)
+            for freq, work in failed_results[cpu].iteritems():
+                msg += '\n\t{}Hz -> {} events'.format(freq, work)
```
Maybe say '\n\t{}Hz -> processed {} events' to clarify that the "event" count is a measure of work throughput.

ionela-voinescu commented:
Looks good to me too. As Valentin I was wondering if there's any point in reporting the full list of performance values for each OPP when it fails, given that it only compares the first and last value for the fail/pass decision. I think it might be worth replacing `Work done did not scale with CPU Freq on CPUs` with `Work at highest OPP wasn't <margin> bigger than work at lowest OPP` just for the user to understand what decides the success/failure of the test.

valschneider commented:
I removed the (freq -> work) list and simply put an error message, as per @ionela-voinescu's comment. Also, since we only compare lowest & highest OPPs, I removed the runs done at OPPs in-between - their results aren't even used anyway.
